### ISSUE TITLE
bug(firestore): logging in deletes any custom fields on user object

### ISSUE THREAD

**Issue by izakfilmalter** [prescottprue/react-redux-firebase#366]
My reactReduxFirebase Config:

```js
    reactReduxFirebase(firebase, {
      userProfile: 'users',
      useFirestoreForProfile: true,
      enableLogging: false,
    })
```

When I use Firestore for my profile, and I write any custom data to it and end up with a profile similare to this:
```
{
  displayName: 'Rick Sanchez',
  email: 'rick@rickandmorty.com',
  membership: {
    QvXyh688YNV29QuhCTXeOXnHt282: true,
    T8Wh9CMHIxT1f9mA5oEETNrOOlt1: true,
  }
}
```

When I logout and log back in my profile looks like this:
```
{
  displayName: 'Rick Sanchez',
  email: 'rick@rickandmorty.com',
}
```

I can get around this by changing my reactReduxFirebase Config to the following, but that is less than optimal:
```js
    reactReduxFirebase(firebase, {
      userProfile: 'users',
      useFirestoreForProfile: true, // Store in Firestore instead of Real Time DB
      enableLogging: false,
      updateProfileOnLogin: false,
    })
```

**Comment by prescottprue** [comment]
@izakfilmalter I was not able to replicate. Which version of `react-redux-firebase` and `redux-firestore` are you running?

Are you setting the custom profile params somewhere else other than on login?

**Comment by izakfilmalter** [comment]
Versions:
```
"react-redux-firebase": "^2.0.0-rc.1",
"redux-firestore": "latest",
```

I'm doing it after a user logs in. Basically it's to manage organization membership within the app. When a user initially logs in, I check if they are apart of an organization, if not, they create one and then I store the orgs id in the membership object. Once that user logs out, and logs back in, it gets wiped. I was using the realtime db before and then it worked fine, once I switched to firestore it started clearing it out.

I am doing this directly via the following call:
```js
    firebase
      .firestore()
      .collection('users')
      .doc(uid)
      .update({
        [`membership.${organizationId}`]: true,
      });
```

**Comment by prescottprue** [comment]
@izakfilmalter Ok, cool. Will try to replicate with that info and get back to you.

Thanks - Its a huge help to report bugs with the Firestore integration since it is still young.

**Comment by jacobfunch** [comment]
I can confirm this is an issue.

**Comment by prescottprue** [comment]
As mentioned by @jacobfunch in #319 it would help to have an option to merge.

**Comment by prescottprue** [comment]
`updateProfileOnLogin: false` seems to also work for me, which makes sense since the `set` isn't being called.

The change to using `update` is part of the current progress for `v2.0.2`.

**Comment by prescottprue** [comment]
Fixed as part of [`v2.0.2`](https://github.com/prescottprue/react-redux-firebase/releases/tag/v2.0.2). Profile's now include more of the auth data by default as well (including timestamps that are formatted).

Let me know if it doesn't work as folks were expecting.
